kiwi_lint/function_based_views.py

```diff
@@ -0,0 +1,130 @@
+
+from pylint import checkers
+from pylint import interfaces
+
+
+class DjangoViewsVisiter(checkers.BaseChecker):
+    """Base class for visiting only astroid modules which contain django views.
+
+    Derivatives could override `visit_views_module` and/or `leave_views_module` to hook just into
+    the walking of view containing astroid modules.
+
+    Class Attributes:
+
+    Instance Attributes:
+        url_mapping
+        Type :: dict[str, 2-tuple(module_name: str, view_name: str)]
+
+        installed_apps_function_filters
+        Type :: iterable[callable[all_apps: iterable[str]] -> iterable[str]]
+
+            List of callbacks, which will be called for pruning the installed apps,
+            on whose views the checker should be invoked.
+    """
+
+    def __init__(self, *args, main_urls_package=None, filters=None, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.installed_apps_function_filters = filters or []
+        django.setup()
+
+        self.__installed_apps = self._get_django_project_apps()
+        print(self.__installed_apps)
+
+        project_urls = import_module(main_urls_package)
+
+        self.url_mapping = self._get_url_view_mapping(project_urls.urlpatterns)
+        self.url_mapping = self._prune_url_mapping(self.url_mapping, self.__installed_apps)
+        self.view_files = set(module for module, _ in self.url_mapping.values())
+
+    def _get_django_project_apps(self):
+        installed_apps = deepcopy(settings.INSTALLED_APPS)  # do not mutate the original
+
+        for _filter in self.installed_apps_function_filters:
+            # allow void filters that just mutate the list
+            res = list(_filter(installed_apps) or [])
+            installed_apps = res if res else installed_apps
+
+        return installed_apps
+
+    @staticmethod
+    def _get_url_view_mapping(urlpatterns):
+        def helper(urlpatterns, prefix='^', acc=None):
+            if acc is None:
+                acc = OrderedDict()
+
+            for url in urlpatterns:
+                if isinstance(url, URLPattern):
+                    key = prefix + url.pattern.regex.pattern.strip('^')
+                    acc[key] = (url.callback.__module__, url.callback.__name__)
+
+                elif isinstance(url, URLResolver):
+                    helper(url.url_patterns, prefix + url.pattern.regex.pattern.strip('^$'), acc)
+            return acc
+
+        return helper(urlpatterns)
+
+    @staticmethod
+    def _prune_url_mapping(url_mapping, installed_apps):
+        pruned = url_mapping.__class__()
+        for k, (module, name) in url_mapping.items():
+            if any(module.startswith(app) for app in installed_apps):
+                pruned[k] = (module, name)
+
+        return pruned
+
+    def visit_module(self, module):
+        if module.name in self.view_files:
+            if hasattr(self, 'visit_views_module'):
+                self.visit_views_module(module)
+
+    def leave_module(self, module):
```

Comment:
`leave_module` and `leave_views_module` are not used and we don't seem to need them so remove both of them.